File: 2025/Python/day_1.py
def parse_input():
    # Parse data from txt file
    with open("Inputs/day_1.txt") as input_file:
        for line in input_file:
            splitter = line.split("\n")

            #Assign to each list
            list_1.append(splitter[0])

    return list_1

def part_1(list_1):
    dial_loc = 50
    zero_count = 0

    for i in list_1:
        mov = i[0]
        num = int(i[1:])

        if mov == "L":
            dial_loc = dial_loc - num
        else:
            dial_loc = dial_loc + num

        #Handle rotations 
        if dial_loc < 0:
            while dial_loc < 0:
                dial_loc += 100
        if dial_loc > 99:
            while dial_loc > 99:
                dial_loc -= 100

        if dial_loc == 0:
            zero_count += 1

    return zero_count

def part_2(list_1):
    dial_loc = 50
    zero_count = 0
    pre_loc = 50

    for i in list_1:
        mov = i[0]
        num = int(i[1:])

        while num > 100:
            zero_count += 1
            num -= 100

        if mov == "L":
            dial_loc = dial_loc - num
        else:
            dial_loc = dial_loc + num

        #Handle rotations 
        if dial_loc < 0:
            while dial_loc < 0:
                dial_loc += 100
                if dial_loc != 0 and pre_loc != 0:
                    zero_count += 1
        if dial_loc > 100:
            while dial_loc > 100:
                dial_loc -= 100
                if dial_loc != 0 and pre_loc != 0:
                    zero_count += 1

        if dial_loc == 0:
            zero_count += 1
        if dial_loc == 100:
            zero_count += 1
            dial_loc = 0
    
        pre_loc = dial_loc


    return zero_count


if __name__ == '__main__':
    # Variables
    list_1 = []

    list_1 = parse_input()

    print("Part 1: " + str(part_1(list_1)))
    print("Part 2: " + str(part_2(list_1)))

File: 2025/Rust/aoc_2025/src/day_1.rs
use std::fs;

pub fn day_1_main() {
    part_1();
    part_2();
}


pub fn part_1(){
    //Ingest data
    let file_name = "inputs/day_1.txt";
    let contents = fs::read_to_string(file_name)
        .expect("Failed to read File");

    //Split by each line
    let parts: Vec<&str> = contents.split("\n").collect();

    let mut dial_loc = 50;
    let mut zero_count = 0;

    for i in 0..parts.len()-1{
        let instruct = parts[i];
        let dir = &instruct[0..1];
        let dir_num = &instruct[1..instruct.len()].parse().unwrap();

        if dir == "L"{
            dial_loc = dial_loc - dir_num;
        }else{
            dial_loc = dial_loc + dir_num;
        }

        //Handle rotations 
        if dial_loc < 0{
            while dial_loc < 0{
                dial_loc += 100
            }
        }
        if dial_loc > 99{
            while dial_loc > 99{
                dial_loc -= 100
            }
        }

        if dial_loc == 0{
            zero_count += 1
        }
    }

    println!("The answer to Part 1 is: {}", zero_count);
}

pub fn part_2(){
    //Ingest data
    let file_name = "inputs/day_1.txt";
    let contents = fs::read_to_string(file_name)
        .expect("Failed to read File");

    //Split by each line
    let parts: Vec<&str> = contents.split("\n").collect();

    let mut dial_loc = 50;
    let mut zero_count = 0;
    let mut pre_loc = 50;

    for i in 0..parts.len()-1{
        let instruct = parts[i];
        let dir = &instruct[0..1];
        let mut dir_num: i32 = instruct[1..instruct.len()].parse().unwrap();

        while dir_num >= 100{
            zero_count += 1;
            dir_num -= 100;
        }

        if dir == "L"{
            dial_loc = dial_loc - dir_num;
        }else if dir == "R"{
            dial_loc = dial_loc + dir_num;
        }

        //Handle rotations 
        if dial_loc < 0{
            while dial_loc < 0{
                dial_loc += 100;
                if dial_loc != 0 && pre_loc != 0{
                    zero_count += 1;
                }
            }
        }
        if dial_loc > 100{
            while dial_loc > 100{
                dial_loc -= 100;
                if dial_loc != 0 && pre_loc != 0{
                    zero_count += 1;
                }
            }
        }

        if dial_loc == 0{
            zero_count += 1;
        }
        if dial_loc == 100{
            zero_count += 1;
            dial_loc = 0;
        }

        pre_loc = dial_loc;

    }

    println!("The answer to Part 2 is: {}", zero_count);
}
